### ISSUE TITLE
Make Index and Data into one file

### ISSUE THREAD

**Issue by luyuncheng** [nmslib/nmslib#467]
Is there any possible to make index and data into one file, when use non optimized spaces.
Because any non optimized spaces, we want to use it, must write and read bin data and index data. 
This can ignore the parameter in the interface which save_data=True, `WriteObjectVectorBinData` and `ReadObjectVectorFromBinData`.

**Comment by searchivarius** [comment]
@luyuncheng I plan to simplify this, but it's not easy for the current version, sorry. That's there's no quick-and-easy fix. BTW, which space are you using?

**Comment by luyuncheng** [comment]
> @luyuncheng I plan to simplify this, but it's not easy for the current version, sorry. That's there's no quick-and-easy fix. BTW, which space are you using?

I see you are planning to add L1 into optimized index #456 that means L1 would save into one file.
but i am using the bit Hamming, is there any possible just store the data into one file even without optimized.

**Comment by searchivarius** [comment]
@luyuncheng ehhh, actually, I think I can make it "optimized" too, probably, even without much trouble. Do you think any other interesting spaces need to be optimized? Currently, it is going to be **dense** l1, l2, linf, cosine, and the inner product.

**Comment by searchivarius** [comment]
@luyuncheng yes, turns out it is somewhat easier for floating-point data. In principle, I can do the same for bit Hamming (and surely will do with time), but at this point, there is a fat chance that I will break an upcoming release, so I wouldn't do it now.

**Comment by luyuncheng** [comment]
> @luyuncheng yes, turns out it is somewhat easier for floating-point data. In principle, I can do the same for bit Hamming (and surely will do with time), but at this point, there is a fat chance that I will break an upcoming release, so I wouldn't do it now.

i am not sure is it correct, i submit a PR for making index and data into one file in hnsw method even without optimized index

**Comment by searchivarius** [comment]
@luyuncheng thanks a lot, but it will take me some time to review and possibly integrate. I am not even fully done with the recent release.

**Comment by luyuncheng** [comment]
> @luyuncheng thanks a lot, but it will take me some time to review and possibly integrate. I am not even fully done with the recent release.

Th
